### PR TITLE
Add extension tests and improve null checks

### PR DESCRIPTION
**Pull request by samatstariongroup**
## Summary
- Avoid recursion when converting streams to ReadOnlyMemory
- Validate arguments in observable and header extensions
- Cover stream, header, and observable extensions with new unit tests

## Testing
- `dotnet test` *(fails: command not found)*

------
https://chatgpt.com/codex/tasks/task_e_68b5a38400448326aff8c2176c5bb084

### CODE REVIEW

File: Mercurio.Tests/Extensions/BasicDeliverEventArgsExtensionsTestFixture.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    using System.Collections.Generic;
+    using System.Text;
+
+    [TestFixture]
+    public class BasicDeliverEventArgsExtensionsTestFixture
+    {
+        [Test]
+        public void VerifyNullArgumentsThrow()
+        {
```
Comment (antoineatstariongroup) [inline Mercurio.Tests/Extensions/BasicDeliverEventArgsExtensionsTestFixture.cs:34]: Please wrap mutliple Assert.That by a 'Assert.EnterMultipleScope()" block

File: Mercurio.Tests/Extensions/BasicDeliverEventArgsExtensionsTestFixture.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+//  </copyright>
+//  ------------------------------------------------------------------------------------------------
+
+namespace Mercurio.Tests.Extensions
+{
+    using Mercurio.Extensions;
+    using RabbitMQ.Client;
```
Comment (antoineatstariongroup) [inline Mercurio.Tests/Extensions/BasicDeliverEventArgsExtensionsTestFixture.cs:24]: using needs cleanup

File: Mercurio.Tests/Extensions/ObservableExtensionsTestFixture.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                return Task.CompletedTask;
+            }, onCompleted: () => completed = true);
+
+            subject.OnNext(1);
+            subject.OnNext(2);
+            subject.OnCompleted();
+
+            await Task.Delay(10);
+
+            Assert.Multiple(() =>
```
Comment (antoineatstariongroup) [inline Mercurio.Tests/Extensions/ObservableExtensionsTestFixture.cs:62]: New standard is the Assert.EnterMultipleScope
